tools/cellxgene_census_builder/tests/anndata/test_anndata.py

```diff
@@ -232,3 +237,180 @@ def test_AnnDataProxy_indexing(census_build_args: CensusBuildArgs, slices: Any)
 
         adata = adata[slc]
         original_X = original_X[slc]
+
+
+@pytest.mark.parametrize("h5ad_simple", ("csr", "csc", "dense"), indirect=True)
+def test_estimated_density(tmp_path: pathlib.Path, h5ad_simple: str) -> None:
+    with open_anndata(h5ad_simple, base_path=tmp_path.as_posix()) as ad:
+        density = ad.get_estimated_density()
+        assert density == ad[1:].get_estimated_density()
```

Comment:
I don't quite understand the reasoning behind the assert in `line 246`:

1. Are you trying to check some character of the data where if the bulk of the density resides in row 0, then you want the assertion to raise? If that is the case, then density could be concentrated in some other row - then what? OR

2. Are you trying to check the character of the implementation of `get_estimated_density()` function in some way? Perhaps that it doesn't fit to some strange non-linear function? I doubt that is your intent but just stating it since I am just guessing here.
3. Something else entirely?

Reply:
I am trying to confirm that the estimated density is not affected by the choice of slice.  For example, if I pick a particularly dense slice, am I getting a different value than the average density for the entire file?